engine/templates_tests/stepCondition.yaml

```diff
@@ -76,3 +76,31 @@ steps:
               then:
                   this: FOO
               message: changed state matched correctly
+    stepFour:
```

Comment:
Maybe could you patch engine_test.go to make sure that the expected custom_state is correctly applied, so we can validate that your code is working.